snuba/datasets/metrics_messages.py

```diff
@@ -74,6 +81,30 @@ def value_for_counter_message(message: Mapping[str, Any]) -> Mapping[str, Any]:
     return {"metric_type": OutputType.COUNTER.value, "count_value": value}
 
 
+def value_for_gauge_message(message: Mapping[str, Any]) -> Mapping[str, Any]:
+    values = message["value"]
+    assert isinstance(values, Mapping), "expected mapping for gauges"
+
+    for k in values:
+        assert isinstance(
+            values[k], (int, float)
+        ), f"{ILLEGAL_VALUE_IN_GAUGE} {INT_FLOAT_EXPECTED}: {values}"
+
+    # Build the nested values structure Clickhouse expects
+    sum = values["sum"]
+    count = values["count"]
+
+    return {
+        "metric_type": OutputType.GAUGE.value,
+        "gauges_values.min": [values["min"]],
+        "gauges_values.max": [values["max"]],
+        "gauges_values.sum": sum,
+        "gauges_values.count": count,
+        "gauges_values.avg": [sum / count],
```

Comment:
Not sure if count can be 0. But you might want to be safe and protect against division by 0.